File: linked lists/deletion at the beginning/deletion.c
#include"stdio.h"
#include"stdlib.h"

/*

creating program for inserting node in the last of
the list and deleting the item from the beginning
this is simple implementation of queue by
using linked list where the First in First out  approach
is used
*/






void print_list(void);
void insert_node(void);

typedef struct
{
    int data;
    struct Node *next ;
}Node;

Node * create_node(void);

Node *start = NULL;  //creating start for accessing the list

void main()
{
    int n;
    while(1)
    {
        system("cls");
        printf("Do you want to do?\n1. Insert at the end\n2. Delete from the beginning\n3. Display\n4. Exit\n");
        scanf("%d",&n);
        switch(n)
        {

            case 1:
                {

                insert_node();
                break;
                }

            case 2:{
                    delete_node();
                    break;
            }
            case 3:
                print_list();
                break;

            case 4:
                exit(0);
                break;

            default:printf("Invalid selection\n");
            }
            _getch();
        }

}


Node * create_node()  // creating a node and returning its address
{
    Node *n;
    n = (Node *)malloc(sizeof(Node));
    return(n);
}


void insert_node()
{
    Node *temp, *t;
    temp = create_node();
    temp->next = NULL;  // putting null in next for every new node created
    //system("cls");
    printf("Enter a number\n");
    scanf("%d",&temp->data);
    if(start == NULL)
    {
        start = temp;
    }
    else
    {
        t = start;
        while(t->next != NULL)
        {
            t = t->next;
        }
        t->next = temp;  // putting the address of newly created node in the last node thus making it second last
    }
}


void print_list()
{
    Node *p;
    //system("cls");
    printf("Your final linked list is \n");
    p = start;
    if (p == NULL)
    {
        printf("List is empty\n");
    }
    else
    {

    while(p->next != NULL)
    {
        printf("%d\n",p->data);
        p = p->next;
    }
    printf("%d\n",p->data);  // the last node doesn't get printed in the loop
    }                         // because p->next becomes null for the last node and loop ends

}

void delete_node()
{
    Node *d;
    d = start;
    if(d == NULL)
    {
        printf("List is empty\n");
    }
    else
    {
        start = start->next;
        free(d);
    }
}

File: linked lists/deletion at the beginning/deletion.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Aug 27 10:05:09 2019

@author: sandeep
"""

'''
program to insert a node at end and delete node from the head
basically it is implementation of a queue using linked list
where insertion occur at one end i.e. tail and deletion occur 
at head, queue follows the First in First out approach

'''


class Node():
    def __init__(self):
        self.data = None
        self.next = None
        
class LinkedList():
    def __init__(self):
        self.head = None
        
    def insert(self):
        num = int(input("Enter a number: "))
        node = Node()
        node.data = num
        
        if self.head is None:
            self.head = node
            return
        
        last = self.head
        
        while last.next:
            last = last.next
            
        last.next = node
        
    def delete(self):
        last = self.head
        
        if self.head is None:
            print("List underflow\n")
            return
        self.head = last.next
        del last
        
        
    def print_list(self):
        last = self.head
        print("Your final list is:")
        if self.head is None:
            print("List is empty\n")
            return
        
        while last:
            print(last.data)
            last = last.next
            
            
            
my_list = LinkedList()

while True:
    print("\nWhat do you want to do?\n1. Insert at end\n2. Delete at head\n3. Print list\n4. Exit")
    choice = int(input("Enter your choice: "))
    
    if choice is 1:
        my_list.insert()
        
    elif choice is 2:
        my_list.delete()
        
    elif choice is 3:
        my_list.print_list()
        
    elif choice is 4:
        break
    else:
        print("Invalid Choice")
        
        
        
        
        
        
    




















